index.js

```diff
@@ -1,57 +1,439 @@
+import { BaseBrush } from './src/brushes/base_brush.class';
+import { PencilBrush } from './src/brushes/pencil_brush.class';
+import { CircleBrush } from './src/brushes/circle_brush.class';
+import { SprayBrush } from './src/brushes/spray_brush.class';
+import { PatternBrush } from './src/brushes/pattern_brush.class';
+import { FabricObject as Object } from './src/shapes/Object/FabricObject';
+import { Line } from './src/shapes/line.class';
+import { Circle } from './src/shapes/circle.class';
+import { Triangle } from './src/shapes/triangle.class';
+import { Ellipse } from './src/shapes/ellipse.class';
+import { Rect } from './src/shapes/rect.class';
+import { Path } from './src/shapes/path.class';
+import { Polyline } from './src/shapes/polyline.class';
+import { Polygon } from './src/shapes/polygon.class';
+import { Text } from './src/shapes/text.class';
+import { IText } from './src/shapes/itext.class';
+import { Textbox } from './src/shapes/textbox.class';
+import { Group } from './src/shapes/group.class';
+import { ActiveSelection } from './src/shapes/active_selection.class';
+import { Image } from './src/shapes/image.class';
+import { getEnv, getDocument, getWindow, setEnvForTests } from './src/env';
+import { createCollectionMixin } from './src/mixins/collection.mixin';
+import { parseAttributes } from './src/parser/parseAttributes';
+import { parseElements } from './src/parser/parseElements';
+import { getFilterBackend } from './src/filters/FilterBackend';
+import { parseStyleAttribute } from './src/parser/parseStyleAttribute';
+import { parsePointsAttribute } from './src/parser/parsePointsAttribute';
+import { parseTransformAttribute } from './src/parser/parseTransformAttribute';
+import { getCSSRules } from './src/parser/getCSSRules';
+
+import './src/controls/default_controls';
+
+const fabric = {
+  parseElements,
+  parseAttributes,
+  cache,
+  version,
+  iMatrix,
+  createCollectionMixin,
+  StaticCanvas,
+  Canvas,
+  config,
+  loadSVGFromURL,
+  loadSVGFromString,
+  initFilterBackend,
+  Canvas2dFilterBackend,
+  WebGLFilterBackend,
+  runningAnimations,
+  Point,
+  Intersection,
+  Color,
+  Control,
+  Observable,
+  Gradient,
+  Pattern,
+  Shadow,
+  BaseBrush,
+  PencilBrush,
+  CircleBrush,
+  SprayBrush,
+  PatternBrush,
+  Object,
+  Line,
+  Circle,
+  Triangle,
+  Ellipse,
+  Rect,
+  Path,
+  Polyline,
+  Polygon,
+  Text,
+  IText,
+  Textbox,
+  Group,
+  ActiveSelection,
+  Image,
+  controlsUtils,
+  util,
+  filters,
+  getFilterBackend,
+  getEnv,
+  getDocument,
+  getWindow,
+  setEnvForTests,
+  parseStyleAttribute,
+  parsePointsAttribute,
+  parseFontDeclaration,
+  parseTransformAttribute,
+  getCSSRules,
+};
+
+if (typeof exports !== 'undefined') {
+  exports.fabric = fabric;
+} else if (typeof define === 'function' && define.amd) {
+  /* _AMD_START_ */
+  define([], function () {
+    return fabric;
+  });
+} else if (typeof window !== 'undefined') {
+  window.fabric = fabric;
+}
```

Comment:
what about this?
next PR?
this is handled by the umd output in rollup

Reply:
We have to fix exports.
Then this will go away automagically i hope.
Next PR or next 2, or next 3, i m not sure how hard it is.
That issue we had with prototype breaking and i m pointing the finger to different transpiling could be a pandora box of issues